detekt/defs.bzl

```diff
@@ -78,65 +148,31 @@ def _impl(ctx):
-            executable = True,
-            cfg = "exec",
-        ),
-        "srcs": attr.label_list(
-            mandatory = True,
-            allow_files = [".kt", ".kts"],
-            allow_empty = False,
-            doc = "Kotlin source code files.",
-        ),
-        "cfgs": attr.label_list(
-            allow_files = [".yml"],
-            default = [],
-            doc = "[Detekt configuration files](https://detekt.github.io/detekt/configurations.html). Otherwise [the default configuration](https://github.com/detekt/detekt/blob/master/detekt-core/src/main/resources/default-detekt-config.yml) is used.",
-        ),
-        "baseline": attr.label(
-            default = None,
-            allow_single_file = [".xml"],
-            doc = "[Detekt baseline file](https://detekt.github.io/detekt/baseline.html).",
-        ),
-        "html_report": attr.bool(
-            default = False,
-            doc = "Enables / disables the HTML report generation. The report file name is `{target_name}_detekt_report.html`.",
-        ),
-        "_txt_report": attr.bool(
-            default = True,
-        ),
-        "xml_report": attr.bool(
-            default = False,
-            doc = "Enables / disables the XML report generation. The report file name is `{target_name}_detekt_report.xml`. FYI Detekt uses the Checkstyle XML reporting format which makes it compatible with tools like SonarQube.",
-        ),
-        "build_upon_default_config": attr.bool(
-            default = False,
-            doc = "See [Detekt `--build-upon-default-config` option](https://detekt.github.io/detekt/cli.html).",
-        ),
-        "disable_default_rulesets": attr.bool(
-            default = False,
-            doc = "See [Detekt `--disable-default-rulesets` option](https://detekt.github.io/detekt/cli.html).",
-        ),
-        "fail_fast": attr.bool(
-            default = False,
-            doc = "See [Detekt `--fail-fast` option](https://detetk.github.io/detekt/cli.html).",
-        ),
-        "parallel": attr.bool(
-            default = False,
-            doc = "See [Detekt `--parallel` option](https://detekt.github.io/detekt/cli.html).",
-        ),
-        "plugins": attr.label_list(
-            default = [],
-            doc = "[Detekt plugins](https://detekt.github.io/detekt/extensions.html). For example, [the formatting rule set](https://detekt.github.io/detekt/formatting.html). Labels should be JVM artifacts (generated via [`rules_jvm_external`](https://github.com/bazelbuild/rules_jvm_external) work).",
-            providers = [JavaInfo],
-        ),
-    },
+    attrs = _get_default_attributes(is_test_target = False),
+    provides = [DefaultInfo],
+    toolchains = ["@rules_detekt//detekt:toolchain_type"],
+)
+
+detekt_test = rule(
+    implementation = _impl,
+    attrs = _get_default_attributes(is_test_target = True),
     provides = [DefaultInfo],
     toolchains = ["@rules_detekt//detekt:toolchain_type"],
+    test = True,
```

Comment:
Does this work out of the box? I believe when you mark something as `executable` or `test` you need to provide an executable for Bazel to run by writing to `ctx.outputs.executable`.

Reply:
Ah I see there is one here https://github.com/davidkurkov/bazel_rules_detekt/pull/1/files#diff-d7c07eade8964ea97850c37c83b90f69b8b3058475560766b6d40f13aa7a8350R154

Reply:
I might be misunderstanding the question,  but there's an executable that's provided in https://github.com/davidkurkov/bazel_rules_detekt/pull/1/files#diff-d7c07eade8964ea97850c37c83b90f69b8b3058475560766b6d40f13aa7a8350R151-R156

It didn't seem to have an effect on running build with the existing `detekt-build-target` - is there something that I should pay attention to in order to verify it's not having some adverse affect on the `non-test-target`?